src/Plugin.php

```diff
@@ -65,6 +65,28 @@ public static function init() {
 
     add_shortcode('user-login-form', __NAMESPACE__ . '\UserLoginForm::getOutput');
 
+    // Disable self pingbacks.
+    add_action('pre_ping', __CLASS__ . '::pre_ping');
+
+    // Disable pingback functionality and remove x-pingback HTTP header.
+    add_filter('xmlrpc_methods', __CLASS__ . '::xmlrpc_methods');
+    add_filter('wp_headers', __CLASS__ . '::wp_headers');
+
+    // Disables admin discussion options for pingbacks.
+    add_filter('pre_update_default_ping_status', '__return_false');
+    add_filter('pre_option_default_ping_status', '__return_zero');
+    add_filter('pre_update_default_pingback_flag', '__return_false');
+    add_filter('pre_option_default_pingback_flag', '__return_zero');
+
+    // Disables insertion of rel="pingback" tags.
+    add_filter('bloginfo_url', __CLASS__ . '::bloginfo_url', 10, 2);
```

Comment:
This is too late. An empty value will cause an invalid HTTP header to be sent: https://github.com/WordPress/WordPress/blob/9fdbe6538e5718a801b2f4ae57260d2d1db4aae6/wp-includes/class-wp.php#L654

Instead you should filter the `pings_open` check: https://developer.wordpress.org/reference/functions/pings_open/

See also https://github.com/WordPress/WordPress/search?q=pings_open